### PR TITLE
add more notes about Eclipse setup

### PR DESCRIPTION
**Pull request by djschny**
Added notes about "search for nested projects..." and the amount of memory Eclipse uses.

### CODE REVIEW

**Comment by jpountz** [comment]
LGTM

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -76,7 +76,7 @@ Contributing to the Elasticsearch codebase
...
+Make sure you have [Maven](http://maven.apache.org) installed, as Elasticsearch uses it as its build system. Integration with IntelliJ and Eclipse should work out of the box. Eclipse users can automatically configure their IDE by running `mvn eclipse:eclipse` and then importing the project into their workspace: `File > Import > Existing project into workspace` and make sure to select `Search for nested projects...` option as Elasticsearch is a multi-module maven project. Additionally you will want to ensure that Eclipse is using 2048m of heap by modifying `eclipse.ini` accordingly to avoid GC overhead errors. Please make sure the [m2e-connector](http://marketplace.eclipse.org/content/m2e-connector-maven-dependency-plugin) is not installed in your Eclipse distribution as it will interfere with setup performed by `mvn eclipse:eclipse`.
 
 Elasticsearch also works perfectly with Eclipse's [m2e](http://www.eclipse.org/m2e/).  Once you've installed m2e you can import Elasticsearch as an `Existing Maven Project`.
```
Comment (nik9000) [inline CONTRIBUTING.md]: Can you remove this line? I'm pretty sure it doesn't work perfectly with m2e any more. It used to back in 1.3-ish.

Comment (djschny) [inline CONTRIBUTING.md]: I'm using it with m2e just fine and additionally it's actually necessary to my knowledge for dependency resolution, etc.

Comment (nik9000) [inline CONTRIBUTING.md]: `mvn eclipse:eclipse` will do the dependency resolution. I like m2e better but its just not well supported by us.

**Pull request merged**